juju/juju
# Remove supported containers check at agent for machine startup

The `running-in-container` is [removed years ago](https://bugs.launchpad.net/ubuntu/+source/init-system-helpers/+bug/1539016), so this uses `systemd-detect-virt` to detect the environment we're running in.

Also the `setupContainerSupport` seems to have no effect on provisioning machines, so this removes the pre-checks and allows any invalid provisions to be failed dynamically.

## Checklist




## QA steps

This is used in the agent machine startup, so any newly added machine should be provisioned just fine. Also the placement should be checked as well.

```
juju bootstrap lxd
juju add-model
juju deploy ubuntu
juju add-machine
juju deploy ubuntu --to 1
```

```
juju bootstrap aws
juju add-model
juju deploy ubuntu
juju add-machine
juju deploy ubuntu --to 1
```

## Links



**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2056200

**Jira card:** JUJU-5633
